test/test_nn.py

```diff
@@ -8143,6 +8143,23 @@ def test_InstanceNorm3d_general(self, device):
         if self.device_type == 'cuda':
             self._test_InstanceNorm_cuda_half(nn.InstanceNorm3d, input, device)
 
+    def test_instancenorm_raises_error_if_input_channels_is_not_num_features(self, device):
+        for i, instance_norm_cls in enumerate([nn.InstanceNorm1d, nn.InstanceNorm2d, nn.InstanceNorm3d]):
+            for no_batch_dim in (True, False):
+                for affine in (True, False):
```

Comment:
nit: you can parametrize the test instead of doing these for loops:

```python
@parametrize_test("instance_norm_cls", [nn.InstanceNorm1d, nn.InstanceNorm2d, nn.InstanceNorm3d], name_fn=lambda c: c.__name__)
@parametrize_test("no_batch_dim", [True, False])
@parametrize_test("affine", [True, False])
def test_instancenorm_raises_error_if_input_channels_is_not_num_features(self, device, instance_norm_cls, no_batch_dim, affine):
```